server/src/main/java/org/elasticsearch/index/mapper/InferenceModelFieldType.java

```diff
@@ -8,14 +8,28 @@
 
 package org.elasticsearch.index.mapper;
 
+import java.util.Map;
+
 /**
  * Field type that uses an inference model.
  */
-public interface InferenceModelFieldType {
+// TODO: Are there any scenarios where extending SimpleMappedFieldType becomes an issue?
+public abstract class InferenceModelFieldType extends SimpleMappedFieldType {
```

Comment:
I changed `InferenceModelFieldType` because as defined before, you could not call basic `MappedFieldType` methods like `name()` on it, which made it pretty inconvenient to use.

This change is not strictly required, but IMO makes for a cleaner implementation overall. Does anyone see any issues with extending `SimpleMappedFieldType` like this?

Reply:
Given that we were already basically doing this in `MockInferenceFieldType` if feels OK, but I will defer to others here.